python/lsst/pipe/tasks/calibrateImage.py

```diff
@@ -424,12 +426,33 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
             config=self.config.photometry_ref_loader, log=self.log)
         self.photometry.match.setRefObjLoader(photometry_loader)
 
-        outputs = self.run(id_generator=id_generator, **inputs)
-
-        butlerQC.put(outputs, outputRefs)
+        # This should not happen with a properly configured execution context.
+        assert not inputs, "runQuantum got more inputs than expected"
+
+        # Specify the fields that `annotate` needs below, to ensure they
+        # exist, even as None.
+        result = pipeBase.Struct(exposure=None,
+                                 stars_footprints=None,
+                                 psf_stars_footprints=None,
+                                 )
+        try:
+            self.run(exposures=exposures, result=result, id_generator=id_generator)
+        except pipeBase.AlgorithmError as e:
+            error = pipeBase.AnnotatedPartialOutputsError.annotate(
+                e,
+                self,
+                result.exposure,
+                result.psf_stars_footprints,
+                result.stars_footprints,
+                log=self.log
+            )
+            butlerQC.put(result, outputRefs)
+            raise error from e
+
+        butlerQC.put(result, outputRefs)
```

Comment:
Should this be in a `finally` block?

Reply:
No. We only `butler.put` if the task succeeded, or if `AlgorithmError` is caught, all other errors can't be annotated, so writing partial outputs is unhelpful.